### PR TITLE
[Concurrency] Task can have instances, add UnsafeCurrentTask; remove any Time related API

### PR DESCRIPTION
**Pull request by ktoso**
- removes anything Deadline related
  -  we're not reviewing those just yet, and for time we'll need to do some more work to be ready here
- `enum Task` -> `struct Task`
- use `CurrentUnsafeTask` as the core of all things
- adds more and better tasks to Task.priority since the existing test didn't really do much

### CODE REVIEW

File: stdlib/public/Concurrency/Task.swift [hunk 1]
```diff
@@ -31,20 +41,52 @@ import Swift
...
+// ==== Current Task -----------------------------------------------------------
+
+extension Task {
+
+  /// Returns 'current' `Task` instance, representing the task from within which
+  /// this function was called.
+  ///
+  /// All functions available on the Task
+  // TODO: once Kavon's async properties land make this computed property
+  public static func current() async -> Task {
```
Comment (DougGregor) [inline stdlib/public/Concurrency/Task.swift:62]: Please use `Builtin.getCurrentAsyncTask()`, because `unsafeCurrent` is not yet implemented.

**Review by DougGregor — APPROVED** [review]

**Comment by ktoso** [comment]
Will do, thanks Doug!

File: stdlib/public/Concurrency/Task.swift (hunk 1 above)
Comment (ktoso) [reply stdlib/public/Concurrency/Task.swift]: Done, not implemented APIs now have `  @available(*, deprecated, message: "Not implemented yet, use Builtin.getCurrentAsyncTask() or Task.___unsafeCurrentAsync() until this function is implemented.")` and we're using the Task.___unsafeCurrentAsync() in tests etc for now.

The changes: https://github.com/apple/swift/pull/35704/commits/9e18cbd217fcfb6d66b646f9ca6212bdd0c588ca

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please smoke test and merge

**Comment by ktoso** [comment]
@swift-ci please smoke test and merge

**Comment by compnerd** [comment]
This seems to have broken the Windows CI.  @shahmishal - seems that the Windows tests didn't trigger here?

**Comment by compnerd** [comment]
https://ci-external.swift.org/job/oss-swift-windows-x86_64-vs2019/4327/consoleText

**Pull request merged**